enhancements/observability/cluster-observability-operator-logging-stack.md

```diff
@@ -0,0 +1,374 @@
+
+### Cluster Obserbability Operator design impact managing OLM-based operators
+
+When an operator (i.e. COO) managing the installation/upgrade of further operators (i.e. CLO and LO) as well as resources owned by the managed operators, they key consideration should be on when managed CRDs will registered on the Kube APIserver. The registration event is a cornerstone for the kubernetes clientsets used in COO to reconcile shared custom resources such as `ClusterLogging` and `Lokistack`. In particular it is mandatory for client abstration such as `informers` and `listers` to execute **only** on resource known by the Kube APIserver.
+
+Therefore the Cluster Observability Operators requires a major design overhaul that supports the following aspects:
+1. Independent reconciliation loops for installing/upgrading managed operators.
+2. Reconciliation loops for shared resources such as `ClusterLogging` and `Lokistack` only and only if registration of the corresponding CRD successful
+3. Robust signaling between the above two and in turn mindful setup and tear-down of underlying kubernetes clientsets/informers/listers.
+
+In summary the Cluster Observability Operator binary will hold two actively communicating sub-controllers each managing distict aspects for an owned resource like `LoggingStack`, i.e.:
+1. **Operators-Controller**: This controller reconciles a `LoggingStack` resource into `OperatorGroup` and `Subscription` resources for CLO and LO only. It tracks success as well as failure conditions for each operator installation/upgrade phase on the `LoggingStack` status field.
+2. **Components-Controller**: This controller reconciles the same `LoggingStack` resource into the resources `ClusterLogging`, `LoggingStack` and `ObservabilityUIPlugin` that in turn install and manage the logging stack components (e.g. collection, storage and visualization plugin).
+
+### LoggingStack status conditions
+
+To effectively track the success/failure status at any point in time of the managed operators (i.e. CLO and LO), the following conditions will be aggregated from OLM resources into the `LoggingStack` status conditions:
+1. TBD
+2. ...
+
+To effectively track the success/failure status of the managed but shared custom resources, the following conditions will be aggregated from the `ClusterLogging` and `LokiStack` resources into the `LoggingStack` status conditions:
+1. TBD
+2. ...
+
+Finally the following conditions will be managed in the `LoggingStack` status conditions about visulization and correlation:
+1. TBD
+2. ...
+
+### Automated korrel8r deployment management
+
+A separate controller namely `correlation_controller` will be introduced in the Cluster Obserbability Operator watching only on create and delete events of `MonitoringStack` and `LoggingStack` resources. If both exist the [korrel8r](korrel8r) deployment will be fully automated on the Cluster Observability Operator installation namespace.
```

Comment:
I'd say that CMO + (LoggingStack|Tracing) is enough to trigger the deployment of korrel8r. Or maybe it should even be always on.